From 2f3267bc017a420854dc98aac1526fb8f11b1874 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Wed, 4 Mar 2015 12:44:39 -0500
Subject: [PATCH] int10: Fix mapping the interrupt vector

pci_device_map_legacy returns 0 on success, and an errno value on
failure.  It works a lot better if we don't treat 0 as failure.

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 hw/xfree86/int10/generic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/xfree86/int10/generic.c b/hw/xfree86/int10/generic.c
index 012d194..8d5c4da 100644
--- w/xfree86/int10/generic.c
+++ hw/xfree86/int10/generic.c
@@ -104,7 +104,7 @@ readIntVec(struct pci_device *dev, unsigned char *buf, int len)
 {
     void *map;
 
-    if (!pci_device_map_legacy(dev, 0, len, 0, &map))
+    if (pci_device_map_legacy(dev, 0, len, 0, &map))
         return FALSE;
 
     memcpy(buf, map, len);
-- 
1.9.3

